scripts/bench_literals.go

```diff
@@ -0,0 +1,164 @@
+package main
+
+import (
+	_ "embed"
+	"fmt"
+	"log"
+	"os"
+	"os/exec"
+	"path"
+	"path/filepath"
+	"reflect"
+	"runtime"
+	"strconv"
+	"strings"
+
+	"mvdan.cc/garble/internal/literals"
+)
+
+const (
+	minDataLen      = 8
+	maxDataLen      = 64
+	stepDataLen     = 4
+	dataCountPerLen = 10
+	moduleName      = "test/literals"
+	garbleSeed      = "o9WDTZ4CN4w"
+
+	// For benchmarking individual obfuscators, we use package=obfuscator mapping
+	// and add a prefix to package name to make sure there are no collisions with system packages.
+	packagePrefix = "literals_bench_"
+)
+
+func generateRunSrc() string {
+	var sb strings.Builder
+
+	sb.WriteString(`
+var alwaysFalseFlag = false
+
+func noop(i interface{}) {
+	if alwaysFalseFlag {
+		println(i)
+	}	
+}
+
+func Run() {
+`)
+
+	dataStr := strings.Repeat("X", maxDataLen)
+	dataBytes := make([]string, maxDataLen)
+	for i := 0; i < len(dataBytes); i++ {
+		dataBytes[i] = strconv.Itoa(i)
+	}
+
+	for dataLen := minDataLen; dataLen <= maxDataLen; dataLen += stepDataLen {
+		for y := 0; y < dataCountPerLen; y++ {
+			fmt.Fprintf(&sb, "\tnoop(%q)\n", dataStr[:dataLen])
+			fmt.Fprintf(&sb, "\tnoop([]byte{%s})\n", strings.Join(dataBytes[:dataLen], ", "))
+		}
+	}
+
+	sb.WriteString("}\n")
+	return sb.String()
+}
+
+func buildTestGarble(tdir string) string {
+	garbleBin := filepath.Join(tdir, "garble")
+	if runtime.GOOS == "windows" {
+		garbleBin += ".exe"
+	}
+
+	output, err := exec.Command("go", "build", "-tags", "garble_testing", "-o="+garbleBin).CombinedOutput()
+	if err != nil {
+		log.Fatalf("garble build failed: %v\n%s", err, string(output))
+	}
+
+	return garbleBin
+}
+
+func writeDateFile(tdir, obfName, src string) error {
+	pkgName := packagePrefix + obfName
+
+	dir := path.Join(tdir, pkgName)
+	if err := os.MkdirAll(dir, 0o777); err != nil {
+		return err
```

Comment:
this is just a benchmark, so I personally wouldn't bother with proper error handling with returning errors or log.Fatalf. you could declare a

```
func handle(err error) { if err != nil { panic(err) }
```

and then lazily handle all errors like

```
handle(os.MkdirAll(dir, 0o777))
```

this is usually too hacky and potentially expensive for regular Go programs, but for a program that just generates and runs a benchmark, we don't really care. making the code easier to read and write is IMO better than making the errors pretty.